src/main/java/net/fabricmc/fabric/mixin/events/tick/MixinPlayerEntity.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2016, 2017, 2018 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.events.tick;
+
+import net.fabricmc.fabric.api.event.player.PlayerTickCallback;
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.world.World;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+@Mixin(PlayerEntity.class)
+public abstract class MixinPlayerEntity extends LivingEntity {
```

Comment:
There is no reason to extend `LivingEntity` here, this can just be a superclass of `Object`

Reply:
Except it is recommended to extend the target's superclass to avoid wasting time in hierarchy validation. (https://github.com/SpongePowered/Mixin/wiki/About-Hierarchy-Validation-in-Mixins)

Reply:
Huh.. my bad, I didn't realise class hierarchy was actually utilized here for validation. I should go and update my own mixins...

Reply:
To be fair, `Object` is the second most optimized thing to extend so it's not a big deal.